OpenRA.Game/Traits/World/ActorMap.cs

```diff
@@ -376,6 +376,11 @@ static void RemoveInfluenceInner(ref InfluenceNode influenceNode, Actor toRemove
 
 		public void Tick(Actor self)
 		{
+			UpdateTriggers();
+		}
+
+		public void UpdateTriggers()
```

Comment:
Updating *all* triggers every time *one* actor changes is a bad idea for a couple of reasons: duplicated/redundant effort, and inconsistent run orders.

Instead, you should make this method set a dirty flag that is then checked during Tick, keeping the real update there.